samtranslator/translator/translator.py

```diff
@@ -125,6 +125,7 @@ def translate(self, sam_template, parameter_values, feature_toggle=None, passthr
         shared_api_usage_plan = SharedApiUsagePlan()
         document_errors = []
         changed_logical_ids = {}
+        route53_record_set_groups = {}
```

Comment:
The key is the logical IDs of the generated `AWS::Route53::RecordSetGroup` resources. For both case of Http API and Rest API, the logical ID is generated based on either `HostedZoneId` or `HostedZoneName`. So, corresponding record set would still be appended into an existing generated record set group instead of overwriting it. 
I added a test for this mixed api case.